fenzo-core/src/main/java/com/netflix/fenzo/AutoScaler.java

```diff
@@ -205,7 +211,29 @@ private void processScalingNeeds(HostAttributeGroup hostAttributeGroup, Concurre
         int excess = hostAttributeGroup.shortFall>0? 0 : hostAttributeGroup.idleHosts.size() - rule.getMaxIdleHostsToKeep();
         int inactiveIdleCount = hostAttributeGroup.idleInactiveHosts.size();
 
-        if ((inactiveIdleCount > 0 || excess > 0) && shouldScaleDown(now, prevScalingActivity, rule)) {
+        if(inactiveIdleCount > 0 && shouldScaleDownInactive(now, prevScalingActivity, rule)) {
+            ScalingActivity scalingActivity = scalingActivityMap.get(rule.getRuleName());
+            long lastReqstAge = (now - scalingActivity.inactiveScaleDownRequestedAt) / 1000L;
+            if(delayScaleDownBySecs>0L && lastReqstAge > 2 * delayScaleDownBySecs) { // reset the request at time
+                scalingActivity.inactiveScaleDownRequestedAt = now;
+            }
+            else if(delayScaleDownBySecs == 0L || lastReqstAge > delayScaleDownBySecs) {
+                scalingActivity.inactiveScaleDownRequestedAt = 0L;
+                scalingActivity.inactiveScaleDownAt = now;
+                Map<String, String> hostsToTerminate = getInactiveHostsToTerminate(hostAttributeGroup.idleInactiveHosts);
+                StringBuilder sBuilder = new StringBuilder();
+                for (String host : hostsToTerminate.keySet()) {
+                    sBuilder.append(host).append(", ");
+                }
+                logger.info("Scaling down inactive hosts " + rule.getRuleName() + " by "
+                        + hostsToTerminate.size() + " hosts (" + sBuilder.toString() + ")");
+                callback.call(
+                        new ScaleDownAction(rule.getRuleName(), hostsToTerminate.values())
```

Comment:
There may be an advantage to making only one callback from this autoscale trigger instead of potentially making two - 1. for inactive VMs, and 2. for active VMs. Since the autoscale trigger runs asynchronous to the main scheduling loop, and we expect a relatively longer time spent in the callback, we may want to combine the two callbacks here into one. That way, the state of the world hasn't changed in between the two callbacks from the same autoscale invocation.